docs/advanced-features/i18n-routing.md

```diff
@@ -204,6 +204,30 @@ export default function IndexPage(props) {
 }
 ```
 
+Note that when using [dynamic routes](/docs/routing/dynamic-routes) you will need to pass the `href` argument of `router.push` as an object.
+```jsx
+import { useRouter } from 'next/router'
+
+export default function IndexPage(props) {
+  const router = useRouter()
+
+  return (
+    <div
+      onClick={() => {
+        router.push({
+          pathname: '/blog/[post]',
+          query: {
+            post: 'hello-world'
+          }
+        }, '/post/hello-world', { locale: 'fr' })
+      }}
+    >
+      to /fr/post/hello-world
+    </div>
+  )
```

Comment:
```suggestion
Note to handle switching just the `locale` while preserving all routing information ([dynamic route](/docs/routing/dynamic-routes.md) query values or hidden href query values) you can provide the `href` as an object:
```jsx
import { useRouter } from 'next/router'
const router = useRouter()
const { pathname, asPath, query } = router
// change just the locale and maintain all other route information including href's query
router.push({ pathname, query }, asPath, { locale: nextLocale });
```

Let's shorten this snippet a bit and tweak the note to make this more general instead of just for dynamic routes as it applies for other case as well.